rust/agents/relayer/src/relayer.rs

```diff
@@ -318,6 +319,18 @@ impl BaseAgent for Relayer {
             .instrument(info_span!("Relayer server"));
         tasks.push(server_task);
```

Comment:
thinking a little about how the broadcast receiver works, I'm realizing that the way we use `MpmcChannel` for the relayer server may mean that we don't clean up the MessageRetryRequests even when they're served (until we have `ENDPOINT_MESSAGES_QUEUE_SIZE = 1_000` of them). This is because we only ever use `receiver` as something to clone here https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/hyperlane-core/src/types/channel.rs#L31. So when we call `run_destination_submitter` on line 344 for each destination chain, we end up creating `n` new receivers, for a total of n+1 receivers. We read from the new `n` receivers in the destination tasks but never the original reciever that lives in the MpmcChannel

Imo this low prio & urgency to fix and doesn't need to be in this PR - maybe something good in a small separate PR or we can put it in the backlog? Will defer to you whether you think it's worth doing rn

Reply:
fixed, that's a great point and explains why there isn't an `mpmc` channel in tokio